Vic2ToHoI4/Source/HOI4World/Ideas/Ideas.cpp

```diff
@@ -37,35 +13,39 @@ HoI4::Ideas::Ideas() noexcept
 
 void HoI4::Ideas::importIdeologicalIdeas()
 {
-	registerKeyword(std::regex("[a-zA-Z_]+"), [this](const std::string& ideology, std::istream& theStream){
+	registerRegex("[a-zA-Z_]+", [this](const std::string& ideology, std::istream& theStream) {
 		ideologicalIdeas.insert(make_pair(ideology, IdeaGroup(ideology, theStream)));
 	});
 
 	parseFile("ideologicalIdeas.txt");
+	clearRegisteredKeywords();
 }
 
 
 void HoI4::Ideas::importGeneralIdeas()
 {
-	clearRegisteredKeywords();
-	registerKeyword(std::regex("[a-zA-Z_]+"), [this](const std::string& ideaGroupName, std::istream& theStream){
-		generalIdeas.push_back(std::make_unique<IdeaGroup>(ideaGroupName, theStream));
+	registerKeyword(std::regex("[a-zA-Z_]+"), [this](const std::string& ideaGroupName, std::istream& theStream) {
+		generalIdeas.push_back(IdeaGroup{ideaGroupName, theStream});
 	});
 
 	parseFile("converterIdeas.txt");
+	clearRegisteredKeywords();
 }
 
 
 void HoI4::Ideas::updateIdeas(const std::set<std::string>& majorIdeologies)
 {
-	auto foundGroup = std::find_if(generalIdeas.begin(), generalIdeas.end(), [](auto& theGroup){ return (theGroup->getName() == "mobilization_laws"); });
-	auto serviceByRequirement = (*foundGroup)->getIdea("service_by_requirement");
+	auto foundGroup = std::find_if(generalIdeas.begin(), generalIdeas.end(), [](auto& theGroup) {
```

Comment:
this whole file has a ton of whitespace to be replaces with \ts, and replacement functions that probably are better moved to a separate updater file.